src/shims/posix/fs.rs

```diff
@@ -71,6 +95,11 @@ impl FileDescriptor for io::Stdin {
     fn seek<'tcx>(&mut self, _communicate_allowed: bool, _offset: SeekFrom) -> InterpResult<'tcx, io::Result<u64>> {
         throw_unsup_format!("cannot seek on stdin");
     }
+
+    fn close<'tcx>(&mut self, _communicate_allowed: bool) -> InterpResult<'tcx, io::Result<i32>> {
+        drop(io::stdin());
```

Comment:
I don't think this does anything.

I am not sure if actually closing FD 0 would do anything (presumably this is observable from other programs communicating with this one), but anyway I wouldn't know how to do that in a cross-platform way either. So I'd say `close` should just be a NOP for all the `std*` descriptors.

Reply:
Hm, thinking about it some more... maybe we should make this error. We cannot actually close the stdin/stdout/stderr that Miri exposes to the outside, and silently doing the wrong thing does not seem great.

But I still think it makes sense to move all interactions with `FileHandle` to the trait, so that it becomes a proper abstraction. The trait could have a default impl for `fn close` to avoid having to repeat the error message.

Reply:
This issue is still around in the latest version of the PR: `drop(io::stdin())` does nothing, you might as well remove that line. But for now, it is probably better if we raise an error instead.

Reply:
The problem with making this error of course is that one cannot actually close stdin/stdout/stderr -- but then in Rust there seems to actually not be a way to do that. So until https://github.com/rust-lang/rust/issues/40032 is resolved I think we are stuck here. :/